src/javascript/app_2/Utils/Validator/__tests__/declarative_validation_rules.spec.js

```diff
@@ -0,0 +1,319 @@
+                let mockvalue = `0.${not_valid_charater}001`;
+                expect(rules['barrier'].func(mockvalue)).to.be.false;
+            });
+        });
+    });
+
+    describe('tests that uses options variable', () => {
+        let options;
+        beforeEach('setting up options', () => {
+            options = {
+                allow_empty: true,
+                min: 0,
+                max: 8,
+                type: 'float',
+                decimals: 2,
+                regex: /[a-z]/g,
+                to: '#txt',
+            }
+        });
+
+        describe('tests that use jQuery', () => {
+            beforeEach('setting up sinon stub', () => {
+                $ = sinon.stub();
+                $.withArgs(options.to).returns({
+                    val: sinon.stub().returns('txt')
+                });
+            });
+
+            describe('validCompare', () => {
+                it('expects return false if empty value passed', () => {
+                    expect(rules['compare'].func('', options)).to.be.false;
+                });
+
+                it('expects return true if the value match', () => {
+                    expect(rules['compare'].func('txt', options)).to.be.true;
+                });
+
+                it('expects false if the value is not match', () => {
+                    expect(rules['compare'].func('txt_wrong', options)).to.be.false;
+                });
+            });
+
+            describe('validNotEqual', () => {
+                it('expects return true if empty value passed', () => {
+                    expect(rules['not_equal'].func('', options)).to.be.true;
+                });
+
+                it('expects return false if the value match', () => {
+                    expect(rules['not_equal'].func('txt', options)).to.be.false;
+                });
+
+                it('expects true if the value is not match', () => {
+                    expect(rules['not_equal'].func('txt_wrong', options)).to.be.true;
+                });
+            });
+        });
+
+        describe('.validMin', () => {
+            it('should return true if options.min is null, undefined, or 0', () => {
+                options.min = 0;
+                expect(rules['min'].func('', options)).to.be.true;
+            });
+            it('should return true if value.length >= options.min', () => {
+                options.min = 2;
+                expect(rules['min'].func('00', options)).to.be.true;
+            });
+            it('should return false if value.length < options.min', () => {
+                options.min = 2;
+                expect(rules['min'].func('0', options)).to.be.false;
+            });
+        });
+
+        describe('.validLength', () => {
+            it('should return true if value.length is in boundary', () => {
+                options.min = 1;
+                expect(rules['length'].func('00', options)).to.be.true;
```

Comment:
IMO, boundary means there is `options.max` as well ?

Reply:
Yea, it's already set